ckan/plugins/__init__.py

```diff
@@ -2,5 +2,3 @@
 
 from ckan.plugins.core import *  # noqa: re-export
 from ckan.plugins.interfaces import *  # noqa: re-export
-
-from ckan.plugins import toolkit  # noqa: re-export
```

Comment:
doesn't this make
```python
import ckan.plugins
ckan.plugins.toolkit
```
stop working?

Reply:
Yes, you are right. Taking it back

Reply:
Oops, that's the problem. This particular line creates a lot of circular references. 
We are using `ckan.plugins` a lot(particularly because of interfaces). Which means that `ckan.logic` loads `IAction` and gets `toolkit` loaded. That prevents  `toolkit` from using `ckan.logic.get_action`

Reply:
Do we really have to keep this object inside `ckan.plugins`? I mean, we have `ckan.plugins.toolkit` as a variable inside `ckan.pluins` and `ckan.plugins.toolkit` as a module.. maybe it's a time to get rid of such duplication? And, even though this is a breaking change, the fix for it is quite straightforward and backward compatible. 

Instead of 
```
import ckan.plugins as p
toolkit = p.toolkit
```

developers have to write
```
import ckan.plugins as p
import ckan.plugins.toolkit as toolkit
```

Reply:
There is another option, but it will be a hack. Smaller than previous implementation of toolkit, but still a hack. I think something like this will solve the problem:

```
# ckan/plugins/__init__.py

from ckan.plugins.core import *  # noqa: re-export
from ckan.plugins.interfaces import *  # noqa: re-export

def __getattr__(self, name):
    if name == "toolkit":
        from . import toolkit
        return toolkit
    super().__getattr__(name)
```

But my original intention was to get rid of magic, not to replace it with smaller amount of magic:)

Reply:
PS [circular import can be fixed by importing `ckan.cli.cli` inside `ckan.__init__.py`](https://github.com/ckan/ckan/pull/6538/files#diff-eacc8c68ea0e9371c3294a7133871cd90f9248595c523016d27a39ec848ae6e7R23). And that's what I've done at the moment. But this is not **right** solution neither :)